modules/holodeckb2b-core/src/test/java/org/holodeckb2b/ebms3/handlers/outflow/CreateSOAPEnvelopeHandlerTest.java

```diff
@@ -0,0 +1,138 @@
+import org.apache.axis2.context.MessageContext;
+import org.apache.axis2.engine.Handler;
+import org.apache.log4j.Appender;
+import org.apache.log4j.Level;
+import org.apache.log4j.LogManager;
+import org.apache.log4j.Logger;
+import org.apache.log4j.spi.LoggingEvent;
+import org.holodeckb2b.common.messagemodel.UserMessage;
+import org.holodeckb2b.common.mmd.xml.MessageMetaData;
+import org.holodeckb2b.core.testhelpers.HolodeckB2BTestCore;
+import org.holodeckb2b.core.testhelpers.TestUtils;
+import org.holodeckb2b.ebms3.constants.MessageContextProperties;
+import org.holodeckb2b.ebms3.packaging.Messaging;
+import org.holodeckb2b.ebms3.packaging.SOAPEnv;
+import org.holodeckb2b.ebms3.packaging.UserMessageElement;
+import org.holodeckb2b.interfaces.core.HolodeckB2BCoreInterface;
+import org.holodeckb2b.interfaces.persistency.entities.IUserMessageEntity;
+import org.holodeckb2b.pmode.helpers.Leg;
+import org.holodeckb2b.pmode.helpers.PMode;
+import org.junit.Before;
+import org.junit.BeforeClass;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.ArgumentCaptor;
+import org.mockito.Captor;
+import org.mockito.Mock;
+import org.mockito.junit.MockitoJUnitRunner;
+
+import java.util.List;
+
+import static org.junit.Assert.*;
+import static org.mockito.Mockito.atLeastOnce;
+import static org.mockito.Mockito.verify;
+
+/**
+ * Created at 23:42 29.01.17
+ *
+ * @author Timur Shakuov (t.shakuov at gmail.com)
+ */
+@RunWith(MockitoJUnitRunner.class)
+public class CreateSOAPEnvelopeHandlerTest {
+
+    @Mock
+    private Appender mockAppender;
+    @Captor
+    private ArgumentCaptor<LoggingEvent> captorLoggingEvent;
+
+    private static String baseDir;
+
+    private static HolodeckB2BTestCore core;
+
+    private CreateSOAPEnvelopeHandler handler;
+
+    @BeforeClass
+    public static void setUpClass() {
+        baseDir = CreateSOAPEnvelopeHandlerTest.class.getClassLoader()
+                .getResource("handlers").getPath();
+        core = new HolodeckB2BTestCore(baseDir);
+        HolodeckB2BCoreInterface.setImplementation(core);
+    }
+
+    @Before
+    public void setUp() throws Exception {
+        handler = new CreateSOAPEnvelopeHandler();
+        Logger logger = LogManager.getRootLogger();
+        logger.addAppender(mockAppender);
+        logger.setLevel(Level.DEBUG);
+    }
+
+    @Test
+    public void testDoProcessing() throws Exception {
+        MessageMetaData mmd = TestUtils.getMMD("handlers/full_mmd.xml", this);
+        // Creating SOAP envelope
+        SOAPEnvelope env = SOAPEnv.createEnvelope(SOAPEnv.SOAPVersion.SOAP_12);
+        // Adding header
+        SOAPHeaderBlock headerBlock = Messaging.createElement(env);
+        // Adding UserMessage from mmd
+        OMElement umElement = UserMessageElement.createElement(headerBlock, mmd);
+
+        MessageContext mc = new MessageContext();
+        mc.setFLOW(MessageContext.OUT_FLOW);
+
+        try {
+            mc.setEnvelope(env);
```

Comment:
Setting the envelope here is a bit redundant as it's the handler's task to add the envelope to the message context ;-)